### ISSUE TITLE
Allow a key sequence binding like `j,k` to not prevent user from insert `j`

### ISSUE THREAD

**Issue by dhatch48** [PowerShell/PSReadLine#1701]
## Description of the new feature/enhancement

You can bind a key sequence like `j,k` to a function or script block by `Set-PSReadLineKeyHandler -Key 'j,k' -ViMode Insert -Function ViCommandMode`, but a user cannot insert a character `j` after that.

## Proposed technical implementation details (optional)

Quoted from https://github.com/PowerShell/PSReadLine/issues/1701#issuecomment-664020465 

 In vim, `jk` or `jj` to exit insert mode is super common.  Usually it works like this:

1. You press <kbd>j</kbd>
1. <kbd>j</kbd> is written to the buffer
1. A timer starts
1. If you press the next key in the sequence before the timer is up, the `j` is removed from the buffer and the command fires
1. If the key you press next isn't in the sequence, or the timer is up then it's treated like a literal

**Comment by daxian-dbw** [comment]
@dhatch48 You can do that by `Set-PSReadLineKeyHandler -Key 'j,k' -ViMode Insert -Function ViCommandMode`

**Comment by SeeminglyScience** [comment]
@daxian-dbw I can't get that to work right.  I mean it works, but you can't type <kbd>j</kbd> anymore.

**Comment by daxian-dbw** [comment]
Yeah, `j` will be bound after that and won't be treated as a literal character. I think that's just an example. Key sequences should be chosen to not interfere with the regular input, like `Set-PSReadlineKeyHandler -Chord 'Ctrl+d,Ctrl+c' -Function CaptureScreen`

**Comment by SeeminglyScience** [comment]
> Yeah, `j` will be bound after that and won't be treated as a literal character. I think that's just an example.

Nah `jk` or `jj` (what I use) to exit insert mode is super common.  Usually it works like this:

1. You press <kbd>j</kbd>
1. `j` is written to the buffer
1. A timer starts
1. If you press the next key in the sequence before the timer is up, the `j` is removed from the buffer and the command fires
1. If the key you press next isn't in the sequence, or the timer is up then it's treated like a literal

**Comment by dhatch48** [comment]
> 
> 
> Yeah, `j` will be bound after that and won't be treated as a literal character. I think that's just an example. Key sequences should be chosen to not interfere with the regular input, like `Set-PSReadlineKeyHandler -Chord 'Ctrl+d,Ctrl+c' -Function CaptureScreen`

I think @SeeminglyScience explained it nicely. Many vim users have bindings of jj or jk to exit insert mode because it is so common to switch modes and this way it keeps your fingers on the home row vs. reaching for Esc key every time.

Anyway, I suppose a sequence timer/timeout is needed to make this work.

**Comment by daxian-dbw** [comment]
I see. Thanks for clarifying it.
I have changed the title and PR description to reflect what is really asked for.

**Comment by Drllap** [comment]
This is what I added to my profile to make `jk` go to comand mode:

```
$j_timer = New-Object System.Diagnostics.Stopwatch

Set-PSReadLineKeyHandler -Key j -ViMode Insert -ScriptBlock {
    [Microsoft.PowerShell.PSConsoleReadLine]::Insert("j")
    $j_timer.Restart()
}

Set-PSReadLineKeyHandler -Key k -ViMode Insert -ScriptBlock {
    if (!$j_timer.IsRunning -or $j_timer.ElapsedMilliseconds -gt 1000) {
        [Microsoft.PowerShell.PSConsoleReadLine]::Insert("k")
    } else {
        [Microsoft.PowerShell.PSConsoleReadLine]::ViCommandMode()
        $line = $null
        $cursor = $null
        [Microsoft.PowerShell.PSConsoleReadLine]::GetBufferState([ref]$line, [ref]$cursor)
        [Microsoft.PowerShell.PSConsoleReadLine]::Delete($cursor, 1)
        [Microsoft.PowerShell.PSConsoleReadLine]::SetCursorPosition($cursor-1)
    }
}
```

**Comment by slacksystem** [comment]
> This is what I added to my profile to make `jk` go to comand mode:
> 
> ```
> $j_timer = New-Object System.Diagnostics.Stopwatch
> 
> Set-PSReadLineKeyHandler -Key j -ViMode Insert -ScriptBlock {
>     [Microsoft.PowerShell.PSConsoleReadLine]::Insert("j")
>     $j_timer.Restart()
> }
> 
> Set-PSReadLineKeyHandler -Key k -ViMode Insert -ScriptBlock {
>     if (!$j_timer.IsRunning -or $j_timer.ElapsedMilliseconds -gt 1000) {
>         [Microsoft.PowerShell.PSConsoleReadLine]::Insert("k")
>     } else {
>         [Microsoft.PowerShell.PSConsoleReadLine]::ViCommandMode()
>         $line = $null
>         $cursor = $null
>         [Microsoft.PowerShell.PSConsoleReadLine]::GetBufferState([ref]$line, [ref]$cursor)
>         [Microsoft.PowerShell.PSConsoleReadLine]::Delete($cursor, 1)
>         [Microsoft.PowerShell.PSConsoleReadLine]::SetCursorPosition($cursor-1)
>     }
> }
> ```

Since this is under not-planned, it seems, I'm trying to figure out how to change this so it's jj instead of jk but I can't figure it out. Do you have any ideas how I could do that?

**Comment by Drllap** [comment]
@slacksystem try this:

```
$j_timer = New-Object System.Diagnostics.Stopwatch
Set-PSReadLineKeyHandler -Key j -ViMode Insert -ScriptBlock {
    if (!$j_timer.IsRunning -or $j_timer.ElapsedMilliseconds -gt 1000) {
        [Microsoft.PowerShell.PSConsoleReadLine]::Insert("j")
        $j_timer.Restart()
        return
    }

    [Microsoft.PowerShell.PSConsoleReadLine]::Insert("j")
    [Microsoft.PowerShell.PSConsoleReadLine]::ViCommandMode()
    $line = $null
    $cursor = $null
    [Microsoft.PowerShell.PSConsoleReadLine]::GetBufferState([ref]$line, [ref]$cursor)
    [Microsoft.PowerShell.PSConsoleReadLine]::Delete($cursor-1, 2)
    [Microsoft.PowerShell.PSConsoleReadLine]::SetCursorPosition($cursor-2)
}

```
